src/qtt/utilities/tools.py

```diff
@@ -1,71 +1,99 @@
+import contextlib
 import copy
-from collections import OrderedDict
-from typing import Type
-import dateutil
-import sys
+import datetime
+import functools
+import importlib
+import inspect
+import logging
 import os
-import numpy as np
+import pickle
+import platform
 import pprint
-import matplotlib
+import sys
+import tempfile
+import time
 import uuid
-import logging
-import qcodes
 import warnings
-import functools
-import pickle
-import inspect
-import tempfile
-from itertools import chain
-import scipy.ndimage as ndimage
+from collections import OrderedDict
 from functools import wraps
-import datetime
-import time
-import importlib
-import platform
+from itertools import chain
+from typing import Optional, Type
+
+import dateutil
+import matplotlib
 import matplotlib.pyplot as plt
+import numpy as np
+import qcodes
+import scipy.ndimage as ndimage
 from matplotlib.widgets import Button
-
+from qcodes.data.data_array import DataArray
 from qcodes.data.data_set import DataSet
+from qcodes.plots.qcmatplotlib import MatPlot
+
+import qtt.pgeometry
+from qtt.pgeometry import (mkdirc,  # import for backwards compatibility
+                           mpl2clipboard, tilefigs)
 
 NotGitRepositoryError: Type[Exception]
 
 try:
-    from dulwich.repo import Repo, NotGitRepository
     from dulwich import porcelain
+    from dulwich.repo import NotGitRepository, Repo
     NotGitRepositoryError = NotGitRepository
 except ModuleNotFoundError:
     warnings.warn('please install dulwich: pip install dulwich --global-option="--pure"')
     NotGitRepositoryError = Exception
 
 # explicit import
-from qcodes.plots.qcmatplotlib import MatPlot
 
 try:
     from qcodes.plots.pyqtgraph import QtPlot
 except BaseException:
     pass
 
-from qcodes.data.data_array import DataArray
-
-import qtt.pgeometry
-from qtt.pgeometry import mpl2clipboard
-
-from qtt.pgeometry import tilefigs, mkdirc  # import for backwards compatibility
 
 # do NOT load any other qtt submodules here
 
 try:
-    import qtpy.QtGui as QtGui
     import qtpy.QtCore as QtCore
+    import qtpy.QtGui as QtGui
     import qtpy.QtWidgets as QtWidgets
 except BaseException:
     pass
 
 
 # %%
 
-from typing import Optional
-import contextlib
+def measure_time(txt: str = 'dt:'):
```

Comment:
Why the extra function returning the class? When you would define the class only you can do (from test_tools.py):
```
with measure_time_context('hi') as m:
    time.sleep(.1)
```
and get the same results

(when you agree you may consider to rename the class to measure_time to keep the rest the same)